marklogic-data-hub-central/build-rpm.gradle

```diff
@@ -0,0 +1,55 @@
+ospackage {
+    packageName = 'hub-central'
+    version = "${project.version}"
+    release = 1
+    os = LINUX
+    type = BINARY
+    arch = NOARCH
+    summary = 'Hub Central Application Server'
+    url = 'http://marklogic.com/'
+    vendor = 'MarkLogic Corporation'
+
+    preInstall file('src/rpm/scripts/pre-install.sh')
+    postInstall file('src/rpm/scripts/post-install.sh')
+    preUninstall file('src/rpm/scripts/pre-uninstall.sh')
+    postUninstall file('src/rpm/scripts/post-uninstall.sh')
+
+    requires('systemd')
+
+    user 'hc-runner'
+    permissionGroup 'hc-runner'
+
+    from(war.outputs.files) {
+        rename { String fileName ->
+            fileName.replace("marklogic-data-hub-central-${project.version}", "hub-central")
+        }
+        fileMode 0755
+        into '/opt/hub-central'
+    }
+
+    from('src/rpm/services/hub-central.service') {
+        addParentDirs(false)
+        fileMode 0755
+        into '/etc/systemd/system'
+    }
+
+    from('src/rpm/configs/hub-central.conf') {
+        fileType CONFIG | NOREPLACE
+        fileMode 0644
+        into '/opt/hub-central'
+    }
+
+    from('src/rpm/configs/hub-central.properties') {
+        addParentDirs(false)
+        fileType CONFIG | NOREPLACE
+        fileMode 0644
+        into '/etc/opt/hub-central'
+    }
+}
+
+buildRpm {
```

Comment:
Where does this write the rpm file to?

Reply:
The RPM gets generated under `marklogic-data-hub-central/build/distributions`

Reply:
Can you add a "doLast" block to buildRpm that logs that the rpm was written to "./build/distributions" (don't need to specify the filename, that'll be obvious)? Otherwise a developer testing this has no idea where the file is written to.

Also, when I ran this locally, I noticed that the hyphen in the version name was changed to a tilde: hub-central-5.3~SNAPSHOT-1.noarch.rpm . Do you know what that is? In theory, that won't matter in practice because our real version names don't have hyphens, just curious what's going on.

Reply:
I believe that's because of the RPM naming conventions:
`[name]-[version]-[release].[arch].rpm`
Since `-` is the only way to distinguish between 'version' and 'release' the Redline java library gets rid of all hyphens in the version section.